x-pack/plugins/ml/public/application/model_management/deployment_setup.tsx

```diff
@@ -168,188 +217,390 @@ export const DeploymentSetup: FC<DeploymentSetupProps> = ({
-                defaultMessage="Cannot exceed {max} - the total number of ML processors."
-                values={{ max: totalMlProcessors }}
+                id="xpack.ml.trainedModels.modelsList.startDeployment.adaptiveAllocationEnabledLabel"
+                defaultMessage="Enabled"
               />
-            ) : null
-          }
-        >
-          <EuiFieldNumber
-            disabled={disableThreadingControls}
-            isInvalid={!!errors.numOfAllocations}
-            fullWidth
-            min={1}
-            max={totalMlProcessors}
-            step={1}
-            name={'numOfAllocations'}
-            value={disableThreadingControls ? 1 : numOfAllocation}
+            }
+            checked={!!config.adaptive_allocations?.enabled}
             onChange={(event) => {
-              onConfigChange({ ...config, numOfAllocations: Number(event.target.value) });
+              onConfigChange({
+                ...config,
+                adaptive_allocations: {
+                  ...config.adaptive_allocations,
+                  enabled: event.target.checked,
+                },
+              });
             }}
-            data-test-subj={'mlModelsStartDeploymentModalNumOfAllocations'}
           />
-        </EuiFormRow>
-      </EuiDescribedFormGroup>
+        </EuiDescribedFormGroup>
+      ) : null}
 
-      {threadsPerAllocations !== undefined ? (
+      {!config.adaptive_allocations?.enabled ? (
         <EuiDescribedFormGroup
           titleSize={'xxs'}
           title={
             <h3>
               <FormattedMessage
-                id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationLabel"
-                defaultMessage="Threads per allocation"
+                id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsLabel"
+                defaultMessage="Number of allocations"
               />
             </h3>
           }
           description={
             <FormattedMessage
-              id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationHelp"
-              defaultMessage="Increase to improve inference latency."
+              id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsHelp"
+              defaultMessage="Increase to improve document ingest throughput."
             />
           }
         >
           <EuiFormRow
             label={
               <FormattedMessage
-                id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationLabel"
-                defaultMessage="Threads per allocation"
+                id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsLabel"
+                defaultMessage="Number of allocations"
               />
             }
             hasChildLabel={false}
             isDisabled={disableThreadingControls}
+            isInvalid={!!errors.numOfAllocations}
+            error={
```

Comment:
Would be better to move this error logic out of inline for better readability as well as avoiding nested ternary statements.